pkg/storage/gcs/gcs.go

```diff
@@ -82,39 +78,7 @@ func (d *driver) Volumes() ([]coreapi.Volume, []coreapi.VolumeMount, error) {
 
 func (d *driver) CompleteConfiguration() error {
 	// Apply global config
-	gcfg, err := clusterconfig.Get()
-	if err != nil {
-		return fmt.Errorf("unable to get global config: %s", err)
-	}
-
-	err = retry.RetryOnConflict(retry.DefaultBackoff, func() error {
-		cur := &coreapi.Secret{
-			TypeMeta: metaapi.TypeMeta{
-				APIVersion: coreapi.SchemeGroupVersion.String(),
-				Kind:       "Secret",
-			},
-			ObjectMeta: metaapi.ObjectMeta{
-				Name:      d.Name + "-private-configuration",
-				Namespace: d.Namespace,
-			},
-		}
-		err := sdk.Get(cur)
-		if err != nil {
-			if !errors.IsNotFound(err) {
-				return fmt.Errorf("failed to get secret %s: %s", cur.Name, err)
-			}
-		}
-
-		if cur.StringData == nil {
-			cur.StringData = make(map[string]string)
-		}
-		cur.StringData["STORAGE_GCS_KEYFILE"] = gcfg.Storage.GCS.KeyfileData
-
-		if errors.IsNotFound(err) {
-			return sdk.Create(cur)
-		}
-		return sdk.Update(cur)
-	})
+	gcfg, err := util.GetConfig(d.Name, d.Namespace)
```

Comment:
should be getting this directly from the install-config object

Reply:
updated in next push